### PR TITLE
Improvements for Docker parametrization

### PR DESCRIPTION
**Pull request by m-mohr**
Fixes reports from #391 and #396, but in a bit more flexible way, also fixes the docs.

@waystilos @geospatial-jeff @constantinius What do you think? I'd like this in the next couple of days.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 COPY . .
-RUN \[ "${DYNAMIC_CONFIG}" == "true" \] && sed -i 's/<!-- <script defer="defer" src=".\/config.js"><\/script> -->/<script defer="defer" src=".\/config.js"><\/script>/g' public/index.html
-RUN npm run build
+RUN \[ "${DYNAMIC_CONFIG}" == "true" \] && sed -i "s|<!-- <script defer=\"defer\" src=\"/config.js\"></script> -->|<script defer=\"defer\" src=\"${pathPrefix}config.js\"></script>|g" public/index.html
+RUN npm run build -- --historyMode="${historyMode}" --pathPrefix="${pathPrefix}"
 
 
 FROM nginx:1-alpine-slim
 
-COPY --from=build-step /app/dist /usr/share/nginx/html
```
Comment (constantinius) [inline Dockerfile:14]: Not sure why bash and nano are installed here. The aim of the Dockerfile was to keep it as small as possible. I would understand why bash is installed, but nano does not seem to be used.

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,28 @@
 FROM node:lts-alpine3.18 AS build-step
 ARG DYNAMIC_CONFIG=true
+ARG historyMode="history"
+ARG pathPrefix="/"
 
 WORKDIR /app
+RUN echo ${pathPrefix} > ./pathprefix
```
Comment (constantinius) [inline Dockerfile:7]: I think it is possible to pass [args between buildstages](https://stackoverflow.com/a/53682110) without writing to a file

File: Dockerfile (hunk 1 above)
Comment (m-mohr) [reply Dockerfile:14]: My bad, forgot to remove this after "debugging".

File: Dockerfile (hunk 2 above)
Comment (m-mohr) [reply Dockerfile]: Good pointer, did it as instructed, thanks.

**Review by constantinius — APPROVED** [review]

**Pull request merged**